File: backend/app/api/v1/platform_settings.py
"""
Endpoints para gestión de configuración global de la plataforma.
Solo accesible por superadmin.
"""

from fastapi import APIRouter, Depends
from pydantic import BaseModel
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.models.user import User
from app.core.dependencies import require_superadmin
from app.services.platform_settings_service import (
    get_all_settings,
    bulk_update_settings,
)

router = APIRouter()


class SettingItem(BaseModel):
    id: str
    key: str
    label: str
    category: str
    is_secret: bool
    has_value: bool
    display_value: str
    real_value: str


class BulkUpdateRequest(BaseModel):
    settings: dict[str, str]  # { "google_client_id": "valor", ... }


@router.get("", response_model=list[SettingItem])
def list_platform_settings(
    admin: User = Depends(require_superadmin),
    db: Session = Depends(get_db),
):
    """Lista todas las configuraciones de la plataforma (valores secretos ofuscados)."""
    return get_all_settings(db)


@router.patch("")
def update_platform_settings(
    body: BulkUpdateRequest,
    admin: User = Depends(require_superadmin),
    db: Session = Depends(get_db),
):
    """Actualiza configuraciones de la plataforma. Solo enviar las que cambiaron."""
    count = bulk_update_settings(db, body.settings)
    return {"ok": True, "updated": count, "message": f"{count} configuracion(es) actualizada(s). Reinicia el backend para aplicar cambios."}

File: frontend/lib/api/platform-settings.ts
/**
 * API client para configuración de plataforma (solo superadmin)
 */

import { authFetch } from "@/lib/auth";

const API_URL = "/api/v1";

export interface PlatformSetting {
  id: string;
  key: string;
  label: string;
  category: string;
  is_secret: boolean;
  has_value: boolean;
  display_value: string;
  real_value: string;
}

export async function getPlatformSettings(): Promise<PlatformSetting[]> {
  const res = await authFetch(`${API_URL}/admin/platform-settings`);
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err.detail || "Error al obtener configuración");
  }
  return res.json();
}

export async function updatePlatformSettings(
  settings: Record<string, string>
): Promise<{ ok: boolean; updated: number; message: string }> {
  const res = await authFetch(`${API_URL}/admin/platform-settings`, {
    method: "PATCH",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ settings }),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err.detail || "Error al actualizar configuración");
  }
  return res.json();
}
